# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/win/src/threadpool_unittest.cc
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 3
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/sandbox/win/src/threadpool_unittest.cc
--- a/security/sandbox/chromium/sandbox/win/src/threadpool_unittest.cc	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/sandbox/win/src/threadpool_unittest.cc	Wed Jun 12 11:10:48 2019 +0100
@@ -2,13 +2,13 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include "sandbox/win/src/win2k_threadpool.h"
+
 #include <stdint.h>
 
-#include "sandbox/win/src/win2k_threadpool.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
-void __stdcall EmptyCallBack(void*, unsigned char) {
-}
+void __stdcall EmptyCallBack(void*, unsigned char) {}
 
 void __stdcall TestCallBack(void* context, unsigned char) {
   HANDLE event = reinterpret_cast<HANDLE>(context);
@@ -23,8 +23,8 @@
 
   EXPECT_EQ(0u, thread_pool.OutstandingWaits());
 
-  HANDLE event1 = ::CreateEventW(NULL, FALSE, FALSE, NULL);
-  HANDLE event2 = ::CreateEventW(NULL, FALSE, FALSE, NULL);
+  HANDLE event1 = ::CreateEventW(nullptr, false, false, nullptr);
+  HANDLE event2 = ::CreateEventW(nullptr, false, false, nullptr);
 
   uint32_t context = 0;
   EXPECT_FALSE(thread_pool.RegisterWait(0, event1, EmptyCallBack, &context));
@@ -38,16 +38,16 @@
   EXPECT_TRUE(thread_pool.UnRegisterWaits(this));
   EXPECT_EQ(0u, thread_pool.OutstandingWaits());
 
-  EXPECT_EQ(TRUE, ::CloseHandle(event1));
-  EXPECT_EQ(TRUE, ::CloseHandle(event2));
+  EXPECT_TRUE(::CloseHandle(event1));
+  EXPECT_TRUE(::CloseHandle(event2));
 }
 
 // Test that register and unregister work, part 2.
 TEST(IPCTest, ThreadPoolRegisterTest2) {
   Win2kThreadPool thread_pool;
 
-  HANDLE event1 = ::CreateEventW(NULL, FALSE, FALSE, NULL);
-  HANDLE event2 = ::CreateEventW(NULL, FALSE, FALSE, NULL);
+  HANDLE event1 = ::CreateEventW(nullptr, false, false, nullptr);
+  HANDLE event2 = ::CreateEventW(nullptr, false, false, nullptr);
 
   uint32_t context = 0;
   uint32_t c1 = 0;
@@ -66,8 +66,8 @@
   EXPECT_TRUE(thread_pool.UnRegisterWaits(&c1));
   EXPECT_EQ(0u, thread_pool.OutstandingWaits());
 
-  EXPECT_EQ(TRUE, ::CloseHandle(event1));
-  EXPECT_EQ(TRUE, ::CloseHandle(event2));
+  EXPECT_TRUE(::CloseHandle(event1));
+  EXPECT_TRUE(::CloseHandle(event2));
 }
 
 // Test that the thread pool has at least a thread that services an event.
@@ -76,22 +76,22 @@
   Win2kThreadPool thread_pool;
 
   // The events are auto reset and start not signaled.
-  HANDLE event1 = ::CreateEventW(NULL, FALSE, FALSE, NULL);
-  HANDLE event2 = ::CreateEventW(NULL, FALSE, FALSE, NULL);
+  HANDLE event1 = ::CreateEventW(nullptr, false, false, nullptr);
+  HANDLE event2 = ::CreateEventW(nullptr, false, false, nullptr);
 
   EXPECT_TRUE(thread_pool.RegisterWait(this, event1, TestCallBack, event2));
 
-  EXPECT_EQ(WAIT_OBJECT_0, ::SignalObjectAndWait(event1, event2, 5000, FALSE));
-  EXPECT_EQ(WAIT_OBJECT_0, ::SignalObjectAndWait(event1, event2, 5000, FALSE));
+  EXPECT_EQ(WAIT_OBJECT_0, ::SignalObjectAndWait(event1, event2, 5000, false));
+  EXPECT_EQ(WAIT_OBJECT_0, ::SignalObjectAndWait(event1, event2, 5000, false));
 
   EXPECT_TRUE(thread_pool.UnRegisterWaits(this));
   EXPECT_EQ(0u, thread_pool.OutstandingWaits());
 
   EXPECT_EQ(static_cast<DWORD>(WAIT_TIMEOUT),
-            ::SignalObjectAndWait(event1, event2, 1000, FALSE));
+            ::SignalObjectAndWait(event1, event2, 1000, false));
 
-  EXPECT_EQ(TRUE, ::CloseHandle(event1));
-  EXPECT_EQ(TRUE, ::CloseHandle(event2));
+  EXPECT_TRUE(::CloseHandle(event1));
+  EXPECT_TRUE(::CloseHandle(event2));
 }
 
 }  // namespace sandbox